src/mpid/ch4/shm/posix/eager/include/posix_eager.h

```diff
@@ -13,9 +13,10 @@
 typedef int (*MPIDI_POSIX_eager_init_t) (int rank, int size);
 typedef int (*MPIDI_POSIX_eager_finalize_t) (void);
 
-typedef int (*MPIDI_POSIX_eager_send_t) (int grank,
-                                         MPIDI_POSIX_am_header_t ** msg_hdr,
-                                         struct iovec ** iov, size_t * iov_num);
+typedef int (*MPIDI_POSIX_eager_send_t) (int grank, const void *buf, MPI_Count count,
+                                         MPI_Datatype datatype, const void *am_hdr,
+                                         MPI_Aint am_hdr_sz, MPIDI_POSIX_am_header_t ** msg_hdr,
+                                         MPI_Aint offset, MPI_Aint * bytes_sent);
```

Comment:
This is a nitpick: since the order of packing is `msg_hdr`, followed by `am_hdr`, followed by data, I find it more intuitive the arrange the parameters follow this order. It also make more sense having `offset` right after datatype since the offset is only associated with data.